src/python/pants/backend/terraform/tool.py

```diff
@@ -407,10 +407,26 @@ def append_only_caches(self) -> dict[str, str]:
 
 
 @dataclass(frozen=True)
-class TerraformProcess:
-    """A request to invoke Terraform."""
+class TerraformCommand:
+    """A single Terraform command to invoke (eg `terraform init`)"""
 
     args: tuple[str, ...]
+
+    def to_args(self, chdir):
+        return (
+            "__terraform/terraform",
+            f"-chdir={shlex.quote(chdir)}",
+        ) + self.args
+
+
+@dataclass(frozen=True)
+class TerraformProcess:
+    """A request to invoke Terraform.
+
+    Accepts multiple commands
```

Comment:
Good thing `tf` supports multiple commands, or we'd have no easy solution...

Reply:
I think the alternative would be to somehow capture the TF cache into a digest so it could be shipped across executors